### PR TITLE
Add terms endpoint

### PR DESCRIPTION
**Pull request by ckr123**

### CODE REVIEW

File: domains/auth/Auth.API/UnitTests/Controllers/TestTermsController.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            .ReturnsAsync(() => fakePrivacyPolicy);
+
+        var response = await termsController.Get();
+        var result = response.Result as OkObjectResult;
+        var privacyPolicy = result?.Value as PrivacyPolicy;
+
+        Assert.Equal(fakePrivacyPolicy.Version, privacyPolicy?.Version);
+        Assert.Equal(fakePrivacyPolicy.Terms, privacyPolicy?.Terms);
+        Assert.Equal(fakePrivacyPolicy.Headline, privacyPolicy?.Headline);
+    }
```
Comment (CodeReaper) [inline domains/auth/Auth.API/UnitTests/Controllers/TestTermsController.cs:42]: Long names again :) 


```suggestion
    private readonly Mock<IPrivacyPolicyStorage> storage = new();

    private readonly TermsController termsController;

    public TestTermsController() => termsController = new(storage.Object);

    [Fact]
    public async Task Get_Terms_ReturnsPrivacyPolicy()
    {
        var fakePolicy = new PrivacyPolicy(
            "<html>nice privacy</html>",
            "v1",
            "PrivacyPolicy"
        );
        storage
            .Setup(p => p.Get())
            .ReturnsAsync(() => fakePolicy);

        var response = await termsController.Get();
        var result = response.Result as OkObjectResult;
        var policy = result?.Value as PrivacyPolicy;

        Assert.Equal(fakePolicy.Version, policy?.Version);
        Assert.Equal(fakePolicy.Terms, policy?.Terms);
        Assert.Equal(fakePolicy.Headline, policy?.Headline);
    }
```

File: domains/auth/Auth.API/UnitTests/Repository/TestPrivacyPolicyStorage.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+using System.Threading.Tasks;
+using API.Configuration;
+using API.Repository;
+using Microsoft.Extensions.Options;
+using Moq;
+using Xunit;
+
+namespace Tests.Repository;
+
+public class TestPrivacyPolicyStorage
```
Comment (CodeReaper) [inline domains/auth/Auth.API/UnitTests/Repository/TestPrivacyPolicyStorage.cs:11]: Same deal as before with long names and folder traversing.

**Review by CodeReaper — CHANGES_REQUESTED** [review]

File: domains/auth/Auth.API/API/Repository/PrivacyPolicyStorage.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+    public PrivacyPolicyStorage(IOptions<AuthOptions> authOptions) => this.authOptions = authOptions.Value;
+
+    public async Task<PrivacyPolicy> Get()
+    {
+        var directory = new DirectoryInfo(authOptions.TermsMarkdownFolder);
+
+        var newestFile = directory
+            .GetFiles()
+            .OrderByDescending(f => f.LastWriteTime)
```
[the remark this answers is not in the document]
Comment (ckr123) [reply domains/auth/Auth.API/API/Repository/PrivacyPolicyStorage.cs:20]: This is from the python code. Is it not the same kubernetes process?

File: domains/auth/Auth.API/IntegrationTests/Controllers/TestTermsController.cs [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+[UnitTest]
+public class TestTermsController
+{
+    private readonly WebApplicationFactory<Program> factory;
+
+    private string resources = Directory
+        .GetParent(Directory.GetCurrentDirectory())?
+        .Parent?
+        .Parent?
+        .FullName + "\\resources";
```
[the remark this answers is not in the document]
Comment (ckr123) [reply domains/auth/Auth.API/IntegrationTests/Controllers/TestTermsController.cs:17]: Had a talk with @duizer it doesnt seem like it. Correct me if i am wrong Paul

File: domains/auth/Auth.API/API/Program.cs [hunk 5]
```diff
@@ -13,7 +13,8 @@
 using Serilog.Formatting.Json;
 
 
-[assembly: InternalsVisibleTo("Tests")]
+[assembly: InternalsVisibleTo("UnitTests")]
+[assembly: InternalsVisibleTo("IntegrationTests")]
```
[the remark this answers is not in the document]
Comment (ckr123) [reply domains/auth/Auth.API/API/Program.cs:17]: Where is that updated?

File: domains/auth/Auth.API/API/Program.cs [hunk 6]
```diff
@@ -37,12 +38,20 @@
 
 builder.Services.AddHttpClient();
 
+builder.Services.Configure<AuthOptions>(options =>
+{
+    options.TermsMarkdownFolder = builder
+        .Configuration
+        .GetValue<string>("TERMS_MARKDOWN_FOLDER", string.Empty);
+});
```
[the remark this answers is not in the document]
Comment (ckr123) [reply domains/auth/Auth.API/API/Program.cs:46]: As agreed with @duizer we will name the ENV variable so it is able to be parsed using iptions

File: domains/auth/Auth.API/IntegrationTests/Controllers/TestTermsController.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (ckr123) [reply domains/auth/Auth.API/IntegrationTests/Controllers/TestTermsController.cs:17]: There is no pretty way of doing thos.

File: domains/auth/Auth.API/IntegrationTests/Controllers/TestTermsController.cs [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+        );
+
+    [Fact]
+    public async Task Get_Endpoint_ReturnsSuccess()
+    {
+        var client = factory.CreateClient();
+
+        var response = await client.GetAsync("/auth/terms");
+
+        Assert.True(response.IsSuccessStatusCode);
```
[the remark this answers is not in the document]
Comment (ckr123) [reply domains/auth/Auth.API/IntegrationTests/Controllers/TestTermsController.cs:32]: This test will be removed and be covered in the apptest as agreed with @duizer

**Review by CodeReaper — APPROVED** [review]

**Pull request merged**